src/main/java/uk/gov/hmcts/darts/audio/controller/AudioController.java

```diff
@@ -61,11 +78,28 @@ public ResponseEntity<byte[]> preview(Integer mediaId, String httpRangeList) {
         return StreamingResponseEntityUtil.createResponseEntity(audioMediaFile, httpRangeList);
     }
 
-    @Override
+
+    @RequestMapping(
+        method = RequestMethod.GET,
+        value = "/audio/preview/{media_id}",
+        produces = {"text/event-stream", "application/json+problem"}
+    )
+    @SneakyThrows
     @SecurityRequirement(name = SECURITY_SCHEMES_BEARER_AUTH)
-    public ResponseEntity<Void> addAudio(MultipartFile file, AddAudioMetadataRequest metadata) {
-        audioService.addAudio(file, metadata);
-        return new ResponseEntity<>(HttpStatus.OK);
+    @Authorisation(contextId = MEDIA_ID,
+        securityRoles = {JUDGE, REQUESTER, APPROVER, TRANSCRIBER, TRANSLATION_QA, RCJ_APPEALS},
+        globalAccessSecurityRoles = {JUDGE})
+    public Flux<ResponseEntity<byte[]>> previewAlternative(
```

Comment:
Flux will allow us to manage back pressure by putting everything on a single event loop.  This is probably a good idea in order to scale out (and avoid denial of service) but will quickly get to a point whereby we are synchronously processing each request depending on the backend thread pool. Just calling it out.

Reply:
An alternate would be to add some asynchronous endpoints so that clients can check in (poll) on the progress of the backend operations. That way client connections do not need to be maintained, plus clients are less likely to wait an ever increasing amount of time as more clients wait for a response.

Reply:
that was covered as part of the spike and they didn't want to do that

Reply:
ok i guess that has to be a limitation with any solution we adopt then. If we use an event based SSE approach to keep the connection open then any similar mechanism is going to get to a point whereby we exhaust the download requests on the container.